packages/typespec-ts/test/modularUnit/serializeUtil.spec.ts

```diff
@@ -0,0 +1,1410 @@
+      export function _customGet1Send(
+        context: Client,
+        body: Widget1,
+        options: CustomGet1Options = { requestOptions: {} }
+      ): StreamableMethod<CustomGet1204Response> {
+        return context
+          .path("/customGet1")
+          .get({
+            ...operationOptionsToRequestParameters(options),
+            body: {
+              id: body["id"],
+              weight: body["weight"],
+              color: body["color"],
+              data: serializeWidgetDataUnion(body["data"]),
+            },
+          });
+      }
+      export async function _customGet1Deserialize(
+        result: CustomGet1204Response
+      ): Promise<void> {
+        if (result.status !== "204") {
+          throw createRestError(result);
+        }
+        return;
+      }
+      export async function customGet1(
+        context: Client,
+        body: Widget1,
+        options: CustomGet1Options = { requestOptions: {} }
+      ): Promise<void> {
+        const result = await _customGet1Send(context, body, options);
+        return _customGet1Deserialize(result);
+      }
+      `,
+      true
+    );
+  });
+
+  it("should generate serialize util if there's a special discriminated union variant of model with bytes property", async () => {
+    const tspContent = `
+    @discriminator("kind")
+    union WidgetData {
+      kind0: WidgetData0;
+      kind1: WidgetData1;
+    }
+
+    model WidgetData0 {
+      kind: "kind0";
+      fooProp: string;
+    }
+    
+    model WidgetData1 {
+      kind: "kind1";
+      data: bytes;
+    }
+    
+    model Widget {
+      @key id: string;
+      weight: int32;
+      color: "red" | "blue";
+    }
+    
+    model Widget1 extends Widget {
+      data: WidgetData
+    }
+
+    interface WidgetService {
+      @get @route("customGet1") customGet1(@body body: Widget1): void;
+    }
+    `;
+
+    // to test the generated deserialized utils for union variant of model with datetime properties.
+    const serializeUtil = await emitModularSerializeUtilsFromTypeSpec(
+      tspContent
+    );
+    assert.ok(serializeUtil);
+    assertEqualContent(
+      serializeUtil?.[0]?.getFullText()!,
+      `
+      import { uint8ArrayToString } from "@azure/core-util";
+      import {
+        WidgetData1 as WidgetData1Rest,
+        WidgetData as WidgetDataRest,
+      } from "../rest/index.js";
+      import { WidgetData1, WidgetData } from "../models/models.js";
+      
+      /** serialize function for WidgetData1 */
+      function serializeWidgetData1(obj: WidgetData1): WidgetData1Rest {
+        return { kind: obj["kind"], data: uint8ArrayToString(obj["data"], "base64") };
+      }
+      
+      /** serialize function for WidgetData */
+      export function serializeWidgetDataUnion(obj: WidgetData): WidgetDataRest {
+        switch (obj.kind) {
+          case "kind1":
+            return serializeWidgetData1(obj);
+          default:
+            return obj;
```

Comment:
in this case, kind0 is not a special union variant, we don't need to do anything here, could just return whatever we get.